### PR TITLE
[RESTEASY-2992] Fix the ClientConfigProviderTestJarHelper to work on Windows

### PR DESCRIPTION
**Pull request by Dkafetzis**
Jira issue link https://issues.redhat.com/browse/RESTEASY-2992

### CODE REVIEW

**Comment by Dkafetzis** [comment]
@jamezp I think the format check is stuck on this one.

File: testsuite/integration-tests/src/test/java/org/jboss/resteasy/test/security/ClientConfigProviderTestJarHelper.java
Comment (jamezp) [inline testsuite/integration-tests/src/test/java/org/jboss/resteasy/test/security/ClientConfigProviderTestJarHelper.java:142]: This should use the `TestUtil.isWindows()`. TBH we probably don't even need to test it as it's valid to surround it in quotes for Linux too.

File: testsuite/integration-tests/src/test/java/org/jboss/resteasy/test/security/ClientConfigProviderTestJarHelper.java
Comment (jamezp) [inline testsuite/integration-tests/src/test/java/org/jboss/resteasy/test/security/ClientConfigProviderTestJarHelper.java:156]: Why this different behavior? I understand we have to customize the command to be an array instead of a string, but this looks overly complicated.

File: testsuite/integration-tests/src/test/java/org/jboss/resteasy/test/security/ClientConfigProviderTestJarHelper.java
Comment (jamezp) [inline testsuite/integration-tests/src/test/java/org/jboss/resteasy/test/security/ClientConfigProviderTestJarHelper.java:140]: Why are we doing this?

File: testsuite/integration-tests/src/test/java/org/jboss/resteasy/test/security/ClientConfigProviderTestJarHelper.java
Comment (jamezp) [inline testsuite/integration-tests/src/test/java/org/jboss/resteasy/test/security/ClientConfigProviderTestJarHelper.java:130]: Why not simply:
```java
final var cp = jarPath + File.pathSeparator + System.getProperty("java.class.path");
```

**Review by jamezp — CHANGES_REQUESTED** [review]

**Comment by jamezp** [comment]
> @jamezp I think the format check is stuck on this one.

Sorry @Dkafetzis, it just required an approval to run for some reason.

File: testsuite/integration-tests/src/test/java/org/jboss/resteasy/test/security/ClientConfigProviderTestJarHelper.java
Comment (Dkafetzis) [reply testsuite/integration-tests/src/test/java/org/jboss/resteasy/test/security/ClientConfigProviderTestJarHelper.java:142]: For some reason when the command is provided with the quotes on process builder on Linux it fails to load the class path. I know that Linux normally supports them but something is not right with the way the ProccessBuilder parses the command.

File: testsuite/integration-tests/src/test/java/org/jboss/resteasy/test/security/ClientConfigProviderTestJarHelper.java
Comment (Dkafetzis) [reply testsuite/integration-tests/src/test/java/org/jboss/resteasy/test/security/ClientConfigProviderTestJarHelper.java:130]: Correct, this will be corrected

File: testsuite/integration-tests/src/test/java/org/jboss/resteasy/test/security/ClientConfigProviderTestJarHelper.java
Comment (Dkafetzis) [reply testsuite/integration-tests/src/test/java/org/jboss/resteasy/test/security/ClientConfigProviderTestJarHelper.java:140]: Originally the above method left a trailing path separator in the new Class Path, after more testing I think it work even with the trailing separator so we can remove this

File: testsuite/integration-tests/src/test/java/org/jboss/resteasy/test/security/ClientConfigProviderTestJarHelper.java
Comment (Dkafetzis) [reply testsuite/integration-tests/src/test/java/org/jboss/resteasy/test/security/ClientConfigProviderTestJarHelper.java]: Apparently Process Builder handles by itself when to put quotes and when not to. I removed the quotes on windows too and the tests passed, so if the tests don't fail here too it is probably ok to remove the extra quotes.

**Comment by Dkafetzis** [comment]
@jamezp changes applied, waiting for another review

**Review by jamezp — CHANGES_REQUESTED** [review]
It looks like some formatting was missed before the commit. Just make sure you compile before committing. You could just do something like the following to fix this:
```bash
mvn clean install -DskipTests
git commit -a --amend
```

**Comment by Dkafetzis** [comment]
@jamezp I don't know why the TCK test is failing now, I also rerun the tests locally and everything is passing

**Review by jamezp — APPROVED** [review]

**Comment by jamezp** [comment]
@Dkafetzis this is a known intermittent failure.

**Pull request merged**
